### PR TITLE
New Feature: Prosemirror Meta Package

### PR DESCRIPTION
**Pull request by bdbch**
### Changes

This PR introduces a new `@tiptap/pm` package which includes all necessary prosemirror modules as a meta package that should resolve the dependencies of Tiptap.

This should resolve issues like #3529 #3497

Right now this PR waits for #3555 since this requires changes to our build process which need to be tested first.

This will allow access to prosemirror packages like this:

```tsx
import { EditorState } from '@tiptap/pm/state'
import * as Tables from '@tiptap/pm/tables'
```

---

### Open Todo

- [x] Check dependencies of tiptap packages & test if pm package resolves requirements correctly @bdbch 
- [x] Resolve local typing issues with `@tiptap/pm`
- [x] Update documentation @bdbch 
- [x] Update installation guides @bdbch
- [x] Add upgrade guides @bdbch 
- [x] Resolve build & test issues
- [x] Reevaluate peer dependencies

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -5,86 +5,66 @@ tableOfContents: true
 # Installation
 
 ## Introduction
+
 Tiptap is framework-agnostic and even works with Vanilla JavaScript (if that’s your thing). The following integration guides help you integrating Tiptap in your JavaScript project.
 
 ## Integration guides
+
 <!-- * [CDN](/installation/cdn) -->
```
Comment (svenadlung) [inline docs/installation.md:13]: I think the idea was to reactivate a CDN solution in future. Maybe we keep this?

File: docs/installation.md [hunk 2]
```diff
@@ -5,86 +5,66 @@ tableOfContents: true
...
-# Installation with NPM 6 or lower
-npm install prosemirror-history prosemirror-dropcursor prosemirror-gapcursor
-
-# Installation with yarn
-yarn add prosemirror-history prosemirror-dropcursor prosemirror-gapcursor
-
-# Installation with pnpm
-pnpm install prosemirror-history prosemirror-dropcursor prosemirror-gapcursor
-```
+**Important note on prosemirror dependencies**<br />
```
Comment (svenadlung) [inline docs/installation.md:43]: ~~prosemirror~~ ProseMirror

File: docs/installation.md [hunk 3]
```diff
@@ -5,86 +5,66 @@ tableOfContents: true
...
+We packaged all required prosemirror dependencies into our own `@tiptap/pm` package. This is also required as a base to get the editor running. If you want
```
Comment (svenadlung) [inline docs/installation.md:44]: ~~also~~
~~prosemirror~~ ProseMirror

File: docs/installation.md [hunk 4]
```diff
@@ -5,86 +5,66 @@ tableOfContents: true
...
+to manage those prosemirror dependencies on your own, make sure to remove `@tiptap/pm` from your dependencies and install all peerDependencies by yourself.
```
Comment (svenadlung) [inline docs/installation.md:45]: ~~prosemirror~~ ProseMirror

File: docs/installation.md [hunk 5]
```diff
@@ -5,86 +5,66 @@ tableOfContents: true
...
 ```
 
 ### 3. Initialize the editor
+
 Everything is in place now, so let’s set up the actual editor now. Add the following code to your JavaScript:
 
 ```js
-import { Editor } from '@tiptap/core'
-import StarterKit from '@tiptap/starter-kit'
+import { Editor } from "@tiptap/core";
```
Comment (svenadlung) [inline docs/installation.md:60]: I think we don't use this formatting in other files, do we?

File: packages/pm/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# @tiptap/pm
+
+[![Version](https://img.shields.io/npm/v/@tiptap/pm.svg?label=version)](https://www.npmjs.com/package/@tiptap/pm)
+[![Downloads](https://img.shields.io/npm/dm/@tiptap/pm.svg)](https://npmcharts.com/compare/tiptap?minimal=true)
+[![License](https://img.shields.io/npm/l/@tiptap/pm.svg)](https://www.npmjs.com/package/@tiptap/pm)
+[![Sponsor](https://img.shields.io/static/v1?label=Sponsor&message=%E2%9D%A4&logo=GitHub)](https://github.com/sponsors/ueberdosis)
+
+## Introduction
+
+tiptap is a headless wrapper around [ProseMirror](https://ProseMirror.net) – a toolkit for building rich text WYSIWYG editors, which is already in use at many well-known companies such as _New York Times_, _The Guardian_ or _Atlassian_.
```
Comment (svenadlung) [inline packages/pm/README.md:10]: ~~tiptap~~ Tiptap
we should fix that in all README files

File: packages/pm/README.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+## What is this `pm` package?
+
+The `pm` package is a wrapper package for [ProseMirror](https://ProseMirror.net). This includes all prosemirror packages that are required to run tiptap.
```
Comment (svenadlung) [inline packages/pm/README.md:14]: all ~~prosemirror~~ ProseMirror packages
~~tiptap~~ Tiptap

File: packages/pm/README.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+## Official Documentation
+
+Documentation can be found on the [tiptap website](https://tiptap.dev).
```
Comment (svenadlung) [inline packages/pm/README.md:18]: ~~tiptap~~ Tiptap

File: docs/installation.md (hunk 1 above)
Comment (bdbch) [reply docs/installation.md]: good point

File: docs/installation.md [hunk 9]
```diff
@@ -5,24 +5,33 @@ tableOfContents: true
...
+- [Vanilla JavaScript](/installation/vanilla-javascript)
+- [React](/installation/react)
+- [Next.js](/installation/nextjs)
+- [Vue 3](/installation/vue3)
+- [Vue 2](/installation/vue2)
+- [Nuxt.js](/installation/nuxt)
+- [Svelte](/installation/svelte)
+- [Alpine.js](/installation/alpine)
+- [PHP](/installation/php)
+- [CDN](/installation/cdn)
```
Comment (svenadlung) [inline docs/installation.md]: Please remove again (comment)

**Review by svenadlung — DISMISSED** [review]

**Comment by jmamakeesic** [comment]
@bdbch It seems that `starter-kit` still needs to updated.
I tried importing the following extensions from `starter-kit@2.0.0-beta.212` and I get the following errors:
```
@tiptap/starter-kit doesn't provide @tiptap/pm, requested by @tiptap/core
@tiptap/starter-kit doesn't provide @tiptap/pm, requested by @tiptap/extension-code-block
@tiptap/starter-kit doesn't provide @tiptap/pm, requested by @tiptap/extension-dropcursor
@tiptap/starter-kit doesn't provide @tiptap/pm, requested by @tiptap/extension-gapcursor
@tiptap/starter-kit doesn't provide @tiptap/pm, requested by @tiptap/extension-history
@tiptap/starter-kit doesn't provide @tiptap/pm, requested by @tiptap/extension-horizontal-rule
```

**Pull request merged**